### ISSUE TITLE
Add some documentation for quicklisp packages

### ISSUE THREAD

**Issue by mbrgm** [NixOS/nixpkgs#26747]
I wanted to add the 'pgloader' package (system?) to nixpkgs but had no idea on how to use the whole quicklisp-to-nix complex. Could someone with knowledge in that area please add some documentation or at least give a hint, so I can work my way from there and document my findings?

ping @7c6f434c

**Comment by 7c6f434c** [comment]
1. I have added some minimal `README.txt` and also updated Quicklisp and fixed stuff I should have fixed a couple of months ago. 

2. I have tried adding pgloader… it has a `Makefile` with tests that also wants to build an executable, so it would be nice to use it; but the `Makefile` wants to just download Quicklisp from upstream website during the build and install a private copy of dependencies; that is unfortunate. So far I have added the package (just to pin the dependencies, if nothing else) which fails; what do you know about its build system?

**Comment by 7c6f434c** [comment]
(just to make it explicit — `README.txt` is more of a general direction of what is supposed to be invoked — overrides are in `quicklisp-to-nix-overrides.nix` — and I know it is in a horrible state so please do submit PRs with additions, or ask specific questions; just ping me in PRs, and for this issue I already receive notifications into «Participating» tab)

**Comment by mbrgm** [comment]
Thank you for taking care of this!

1. Really helps understand this part of nixpkgs a bit better, thanks!
2. Unfortunately I don't know anything about lisp (except a tiny bit of elisp), neither about its ecosystem and tools. I just wanted to add pgloader to nixpkgs because I wanted to use it, so I started packaging it as a standalone package, then found that there are some lisp packages in the tree already and then tried to figure out how to add it this way. I'll try to see how far I can get now.

Thank you again!

**Comment by bradleyjensen** [comment]
Using the documentation, I was able to run the quicklisp-to-nix program and generate nix expressions for packages I'm interested in.  However, I found that some of the resulting nix expressions for pre-existing quicklisp-to-nix packages were missing some dependencies.  As in, running quicklisp-to-nix on nixpkgs as it is today (no modifications) results in some packages losing dependency information.  Some packages still had correct dependencies.  I dug into it a bit more and I found that quicklisp itself wasn't listing dependencies correctly.  For example,
`(ql-dist:required-systems (ql-dist:find-system "qmynd")) ; => nil`
But if you look at the [quickdocs](http://quickdocs.org/qmynd/) (or try to install qmynd) you'll find it actually depends on multiple packages.  However, when installing qmynd, I noticed that quicklisp was repeatedly trying to load qmynd and then stopping to install one of its dependencies.  I think quicklisp might be installing dependencies when they are found to be missing rather than installing them upfront.

Have I followed the directions incorrectly, or has quicklisp changed its behaviors?

**Comment by 7c6f434c** [comment]
I am using some of the Quicklisp interfaces to get thje dependencies; in different months (i.e. different monthly releases) it is either correct or misses some dependencies, upstream seems to work on making this more robust. Maybe I should switch to some more robust way of finding out the dependencies, but I haven't yet chosen a strategy.

**Comment by bradleyjensen** [comment]
I believe I greatly simplified the task of using quicklisp to nix in pr #28810. Now it’s just one nix-shell command. The source has more documentation as well.

**Comment by BadDecisionsAlex** [comment]
Subscribing to this.

I just want to use SBCL like it's normal, system wide, with stuff just working :(
Quicklisp and SBCL on their own work alright until you need any libs and then everything blows up.

**Comment by vxid** [comment]
I second this. I would ideally just be able to setup sbcl, quicklisp and any other lisp module that I need in my configuration.nix and have everything available everywhere (slime, stumpwm etc). Did anyone manage to do so?

**Comment by matthew-piziak** [comment]
still important to me

**Comment by 7c6f434c** [comment]
(looking at the first message — I _hope_ that the now-unbreaked pgloader actually works and will stay unbroken)

**Comment by 7c6f434c** [comment]
(for imperative Quicklisp use I guess one could just generate a one-liner to export `$LD_LIBRARY_PATH` or maybe `$NIX_LDL_LIBRARY_PATH` if wrapper is used, generating the path using `lib.makeLibraryPath`)

**Comment by teu5us** [comment]
> (for imperative Quicklisp use I guess one could just generate a one-liner to export `$LD_LIBRARY_PATH` or maybe `$NIX_LDL_LIBRARY_PATH` if wrapper is used, generating the path using `lib.makeLibraryPath`)

As well as CPATH and PKG_CONFIG_PATH (with a C compiler in buildInputs), as some system have parts written in C

**Comment by 7c6f434c** [comment]
BTW, it looks like `nix-shell` sets everything needed when given native libs as build inputs, except for `LD_LIBRARY_PATH` and `CPATH` (looking at the Quicklisp-scraping `shell.nix`)

**Comment by SquircleSpace** [comment]
After trying really hard to use existing nixpkgs infrastructure for satisfying my quicklime dependencies, I eventually gave up and took a fundamentally different approach. Check out https://github.com/SquircleSpace/ql2nix

**Comment by 7c6f434c** [comment]
Yeah, ql2nix is not a perfect fit for Nixpkgs packaging, but nice if you want to have something virtualenv-style but in the store.
